src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperDescriptorCache.cs

```diff
@@ -17,5 +20,117 @@ internal static class TagHelperDescriptorCache
 
         internal static void Set(int hashCode, TagHelperDescriptor descriptor) =>
             s_cachedTagHelperDescriptors.Set(hashCode, descriptor);
+
+#nullable enable
+        internal static int GetTagHelperDescriptorCacheId(TagHelperDescriptor descriptor)
+        {
+            var hash = HashCodeCombiner.Start();
+            hash.Add(descriptor.Kind, StringComparer.Ordinal);
+            hash.Add(descriptor.AssemblyName, StringComparer.Ordinal);
+            hash.Add(descriptor.Name, StringComparer.Ordinal);
+            hash.Add(descriptor.DisplayName, StringComparer.Ordinal);
+            hash.Add(descriptor.CaseSensitive ? 1 : 0);
+
+            CalculateForProperty(ref hash, descriptor.BoundAttributes, CalculateBoundAttributeCacheId);
+            CalculateForProperty(ref hash, descriptor.TagMatchingRules, CalculateTagMatchingRuleCacheId);
+            CalculateForProperty(ref hash, descriptor.AllowedChildTags, CalculateAllowedChildTagsCacheId);
+
+            if (descriptor.Diagnostics != null)
+            {
+                for (var i = 0; i < descriptor.Diagnostics.Count; i++)
+                {
+                    hash.Add(descriptor.Diagnostics[i]);
+                }
+            }
+
+            if (descriptor.Metadata != null)
+            {
+                foreach (var kvp in descriptor.Metadata)
+                {
+                    hash.Add(kvp.Value, StringComparer.Ordinal);
```

Comment:
Odd that the key isn't used here, but I guess if thats how the previous code worked, then who am I to argue 🤷‍♂️

Reply:
You were right to point it out. I copied this from the original PR, but looking at the current source it is taken into account. I've updated it to reflect the current implementation.